pkg/cmd/pipelinerun/log_reader.go

```diff
@@ -121,11 +132,58 @@ func (lr *LogReader) readAvailableLogs(pr *v1alpha1.PipelineRun) (<-chan Log, <-
 
 			pipeLogs(logC, errC, tlr)
 		}
+		if pr.Status.Conditions[0].Status == corev1.ConditionFalse {
+			errC <- fmt.Errorf(pr.Status.Conditions[0].Message)
+		}
 	}()
 
 	return logC, errC, nil
 }
 
+// reading of logs should wait till the status of run is unknown
+// only if run status is unknown, open a watch channel on run
+// and keep checking the status until it changes to true|false
+// or the reach timeout
+func (lr *LogReader) waitUntilAvailable(timeout time.Duration) error {
+	var first = true
+	opts := metav1.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector("metadata.name", lr.Run).String(),
+	}
+	tkn := lr.Clients.Tekton
+	run, err := tkn.TektonV1alpha1().PipelineRuns(lr.Ns).Get(lr.Run, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if run.Status.Conditions[0].Status != corev1.ConditionUnknown {
```

Comment:
check the `status.condition` array length? because in rare conditions could be empty?

Reply:
Done